xhyve/xhyve.go

```diff
@@ -612,42 +612,44 @@ func (d *Driver) publicSSHKeyPath() string {
 	return d.GetSSHKeyPath() + ".pub"
 }
 
-func readLine(path string) string {
-	line := ""
+func readLine(path string) (string, error) {
 	inFile, err := os.Open(path)
 	if err != nil {
-		log.Debugf("Not able to open %s", path)
+		return "", err
 	}
 	defer inFile.Close()
+
 	scanner := bufio.NewScanner(inFile)
-	scanner.Split(bufio.ScanLines)
 	for scanner.Scan() {
-		line = scanner.Text()
-		if kernelOptionRegexp.MatchString(line) {
-			break
+		if kernelOptionRegexp.Match(scanner.Bytes()) {
+			m := kernelOptionRegexp.FindSubmatch(scanner.Bytes())
+			return string(m[1]), nil
```

Comment:
@zchee Have you tested those changes because for me it's not working and looks like this part is not able to get kernel options, may be we need to look scanner.Bytes() stuff.  I will do further test after you make regex changes.

Below you can see there is no extracted kernel options.
```
 ./cmd -D create centos -d xhyve --xhyve-boot2docker-url=file:/Users/prkumar/minishift-centos.iso
Docker Machine Version:  0.8.2, build HEAD
Found binary path at /Users/prkumar/work/github/go_practice/bin/docker-machine-driver-xhyve
Launching plugin server for driver xhyve
Plugin server listening at address 127.0.0.1:49262
() DBG | operation not supported by device
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(flag-lookup) Calling .GetMachineName
(flag-lookup) Calling .DriverName
(flag-lookup) Calling .GetCreateFlags
Found binary path at /Users/prkumar/work/github/go_practice/bin/docker-machine-driver-xhyve
Launching plugin server for driver xhyve
Plugin server listening at address 127.0.0.1:49266
() DBG | operation not supported by device
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(centos) Calling .GetMachineName
(centos) Calling .DriverName
(centos) Calling .GetCreateFlags
(centos) Calling .SetConfigFromFlags
Running pre-create checks...
(centos) Calling .PreCreateCheck
(centos) DBG | ===== Docker Machine xhyve Driver Version 0.3.0 (HEAD) =====
(centos) DBG | 
(centos) DBG | executing: /usr/local/bin/VBoxManage -v
(centos) DBG | STDOUT: 5.0.12r104815
(centos) DBG | 
(centos) DBG | STDERR: 
(centos) Calling .GetConfigRaw
Creating machine...
(centos) Calling .Create
(centos) Downloading /Users/prkumar/.docker/machine/cache/boot2docker.iso from file:/Users/prkumar/minishift-centos.iso...
(centos) Creating VM...
(centos) DBG | Mounting boot2docker.iso
(centos) DBG | executing: &{/usr/bin/hdiutil [hdiutil attach /Users/prkumar/.docker/machine/machines/centos/boot2docker.iso -mountpoint /Users/prkumar/.docker/machine/machines/centos/b2d-image] []  <nil> 0xc42002a010 0xc42002a018 [] <nil> <nil> <nil> <nil> <nil> false [] [] [] [] <nil> <nil>} attach /Users/prkumar/.docker/machine/machines/centos/boot2docker.iso -mountpoint /Users/prkumar/.docker/machine/machines/centos/b2d-image
(centos) /dev/disk27         	                               	/Users/prkumar/.docker/machine/machines/centos/b2d-image
(centos) DBG | Extracting Kernel Options...
(centos) DBG | Extracted Options   
```